source/common/access_log/access_log_formatter.cc

```diff
@@ -23,6 +23,9 @@ static const std::string UnspecifiedValueString = "-";
 
 namespace {
 
+// Matches newline pattern in a StartTimeFormatter format string.
+static const std::regex NEWLINEPATTERN{"%[1-9]*n", std::regex::optimize};
```

Comment:
you'll need a `CONSTRUCT_ON_FIRST_USE` pattern here, to avoid static-initialization order fiasco